src/applets/icon-tasklist/IconButton.vala

```diff
@@ -966,6 +967,39 @@ public class IconButton : Gtk.ToggleButton
             } else {
                 launch_app(event.time);
             }
+        } else if (event.button == 2) { // Middle click
+            GLib.List<unowned Wnck.Window> windows;
+            bool middle_click_create_new_instance = false;
+
+            if (this.settings != null) { // Settings defined
+                middle_click_create_new_instance = this.settings.get_boolean("middle-click-launch-new-instance");
+            }
+
+            if (middle_click_create_new_instance) {
+                if (class_group != null) {
+                    windows = class_group.get_windows().copy();
+                } else {
+                    windows = new GLib.List<unowned Wnck.Window>();
+                }
+    
+                if (windows.length() == 0) {
+                    launch_app(Gtk.get_current_event_time());
+                } else if (this.app_info != null) {
+                    string[] actions = app_info.list_actions();
+    
+                    if ("new-window" in actions) {
+                        preferred_action = "new-window";
+                    }
```

Comment:
I don't think there really is much point in having a global private `preferred_action` variable if we're always going to be checking if is a supported action. We should either check if it is an empty string and only iterate over the actions if it is, or eliminate the variable altogether and just do our launch context in the same if block as our "new-window" in actions, else fallback to launching the app with the current event time. Something like the following:

```
string[] actions = app_info.list_actions();

if ("new-window" in actions) {
	launch_context.set_screen(get_screen());
	launch_context.set_timestamp(Gdk.CURRENT_TIME);
	this.app_info.launch_action("new-window", launch_context);
} else {
	launch_app(Gtk.get_current_event_time());
}
```

Reply:
Done